### PR TITLE
Chore: Convert QueryOperationAction test to RTL

### LINKED ISSUE #51146: Convert `public/app/core/components/QueryOperationRow/QueryOperationAction.test.tsx` to RTL

### PR DESCRIPTION
**Pull request by Seyaji**
Which issue(s) this PR fixes:
Fixes https://github.com/grafana/grafana/issues/51146

### CODE REVIEW

**Comment by Seyaji** [comment]
@ashharrison90 can you create an issue for this please? 🙂

File: public/app/core/components/QueryOperationRow/QueryOperationAction.test.tsx [hunk 1]
```diff
@@ -1,24 +1,53 @@
...
-      expect(actionEl).toHaveLength(1);
-      expect(clickSpy).not.toBeCalled();
+  it('should call on click handler', () => {
+    const clickSpy = jest.fn();
+    setup({ disabled: false, onClick: clickSpy });
+
+    expect(clickSpy).not.toHaveBeenCalled();
+    const queryButton = screen.getByRole('button', { name: selectors.components.QueryEditorRow.actionButton('test') });
+
+    fireEvent.click(queryButton);
```
Comment (ashharrison90) [inline public/app/core/components/QueryOperationRow/QueryOperationAction.test.tsx:37]: can we use `userEvent` here instead? test needs to be made async then as well
```suggestion
    await userEvent.click(queryButton);
```

**Review by ashharrison90 — CHANGES_REQUESTED** [review]
minor change about preferring `userEvent` over `fireEvent`, lgtm otherwise 😄

File: public/app/core/components/QueryOperationRow/QueryOperationAction.test.tsx [hunk 2]
```diff
@@ -1,24 +1,53 @@
...
+
+const setup = (propOverrides?: Partial<QueryOperationActionProps>) => {
+  const props: QueryOperationActionProps = {
+    icon: 'panel-add',
+    title: 'test',
+    onClick: jest.fn(),
+    disabled: false,
+  };
+
+  Object.assign(props, propOverrides);
```
[the remark this answers is not in the document]
Comment (Seyaji) [reply public/app/core/components/QueryOperationRow/QueryOperationAction.test.tsx:16]: Done!

File: public/app/core/components/QueryOperationRow/QueryOperationAction.test.tsx (hunk 1 above)
Comment (Seyaji) [reply public/app/core/components/QueryOperationRow/QueryOperationAction.test.tsx:37]: Sure :)

**Review by ashharrison90 — CHANGES_REQUESTED** [review]
tiny change

File: public/app/core/components/QueryOperationRow/QueryOperationAction.test.tsx [hunk 3]
```diff
@@ -1,24 +1,55 @@
...
+const setup = (propOverrides?: Partial<QueryOperationActionProps>) => {
+  const props: QueryOperationActionProps = {
+    icon: 'panel-add',
+    title: 'test',
+    onClick: jest.fn(),
+    disabled: false,
+    ...propOverrides,
+  };
+
+  Object.assign(props);
```
[the remark this answers is not in the document]
Comment (Seyaji) [reply public/app/core/components/QueryOperationRow/QueryOperationAction.test.tsx:18]: 🙃 Sorted! I should have checked that more carefully

[the remark this answers is not in the document]
Comment (ashharrison90) [reply public/app/core/components/QueryOperationRow/QueryOperationAction.test.tsx]: no problem, that's what code reviews are for 😄

**Review by ashharrison90 — APPROVED** [review]
thanks again for a great contribution 🙇

**Pull request merged**